app/services/repository_actions/repository_cell_resolver.rb

```diff
@@ -0,0 +1,113 @@
+# frozen_string_literal: true
+
+module RepositoryActions
+  class RepositoryCellResolver
+    def initialize(cell, new_row, team)
+      @cell    = cell
+      @new_row = new_row
+      @team    = team
+    end
+
+    def call
+      case @cell.value_type
+      when 'RepositoryListValue'
+        clone_repository_list_value
+      when 'RepositoryTextValue'
+        clone_repository_text_value
+      when 'RepositoryAssetValue'
+        clone_repository_asset_value
+      when 'RepositoryDateValue'
+        clone_repository_date_value
+      end
+    end
+
+    private
+
+    def clone_repository_list_value
+      old_value = @cell.value
+      RepositoryListValue.create(
+        old_value.attributes.merge(id: nil,
+          repository_cell_attributes: {
+            repository_row: @new_row,
+            repository_column: @cell.repository_column
+          })
+      )
+    end
+
+    def clone_repository_text_value
+      old_value = @cell.value
+      RepositoryTextValue.create(
+        old_value.attributes.merge(id: nil,
+          repository_cell_attributes: {
+            repository_row: @new_row,
+            repository_column: @cell.repository_column
+          })
+      )
+    end
+
+    def clone_repository_asset_value
+      old_value = @cell.value
+      new_asset = create_new_asset(old_value.asset)
+      RepositoryAssetValue.create(
+        old_value.attributes.merge(
+          id: nil,
+          asset: new_asset,
+          repository_cell_attributes: {
+            repository_row: @new_row,
+            repository_column: @cell.repository_column
+          }
+        )
+      )
+    end
+
+    def clone_repository_date_value
+      old_value = @cell.value
+      RepositoryDateValue.create(
+        old_value.attributes.merge(id: nil,
+          repository_cell_attributes: {
+            repository_row: @new_row,
+            repository_column: @cell.repository_column
+          })
+      )
+    end
+
+    # reuses the same code we have in copy protocols action
+    def create_new_asset(old_asset)
+      new_asset = Asset.new_empty(
+        old_asset.file_file_name,
+        old_asset.file_file_size
+      )
+      new_asset.created_by = old_asset.created_by
+      new_asset.team = old_asset.team
```

Comment:
Also, team should probably set to `@team`? I mean it shouldn't matter, but maybe it's more safe this way.

Reply:
hmm I don't know you can copy only items in the same repository and that repository belongs to a team so it will be a bit odd that repository items will belong to other team

Reply:
but other way it should be OK one way or another